### PR TITLE
Fix/schema updates

### PR DESCRIPTION
**Pull request by riteshsp2000**
- Update schema as per the latest requirements
- Update types, queries, mutations as the schema

### CODE REVIEW

File: server/schema/mutations/coordinator.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+const Coordinator = require("../../models/coordinator");
+
+// GraphQL Types
+const { CoordinatorType } = require("../types");
+
+const addCoordinator = {
+  type: CoordinatorType,
+  args: {
+    name: { type: new GraphQLNonNull(GraphQLString) },
+    rollNumber: { type: new GraphQLNonNull(GraphQLString) },
```
Comment (abhibhaw) [inline server/schema/mutations/coordinator.js:18]: isActive isn't used and not have any default value as well

File: server/schema/mutations/coordinator.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    contact: { type: new GraphQLNonNull(GraphQLString) },
+    email: { type: new GraphQLNonNull(GraphQLString) },
+    year: { type: GraphQLInt },
+  },
+  async resolve(parent, { name, contact, email, rollNumber, year }) {
+    try {
+      const checkCoordinator = await Coordinator.exists({ rollNumber });
```
Comment (abhibhaw) [inline server/schema/mutations/coordinator.js:25]: Better to make roll number field unique than manual comparison

File: server/schema/mutations/coordinator.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+const removeCoordinator = {
+  type: CoordinatorType,
+  args: {
+    rollNumber: { type: new GraphQLNonNull(GraphQLString) },
+  },
+  async resolve(parent, { rollNumber }) {
+    await Coordinator.findOneAndDelete({ rollNumber }, (error, docs) => {
+      if (error) {
+        console.log(error);
```
Comment (abhibhaw) [inline server/schema/mutations/coordinator.js:91]: deleting with ID makes more sense

File: server/schema/mutations/mentee.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+  args: {
+    name: { type: new GraphQLNonNull(GraphQLString) },
+    rollNumber: { type: new GraphQLNonNull(GraphQLString) },
+    contact: { type: new GraphQLNonNull(GraphQLString) },
+    email: { type: new GraphQLNonNull(GraphQLString) },
+    mentor: { type: new GraphQLNonNull(GraphQLID) },
+  },
+  async resolve(parent, { name, rollNumber, mentor, contact, email }) {
+    try {
+      const checkMentee = await Mentee.exists({ rollNumber });
```
Comment (abhibhaw) [inline server/schema/mutations/mentee.js:26]: Same as coordinator

File: server/schema/mutations/mentee.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+};
+
+const removeMentee = {
+  type: MenteeType,
+  args: {
+    rollNumber: { type: new GraphQLNonNull(GraphQLString) },
+  },
+  async resolve(parent, { rollNumber }) {
+    try {
+      await Mentee.findOneAndDelete({ rollNumber }, (error, docs) => {
```
Comment (abhibhaw) [inline server/schema/mutations/mentee.js:92]: Delete with ID makes no sense

File: server/schema/mutations/mentor.js [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+  args: {
+    name: { type: new GraphQLNonNull(GraphQLString) },
+    rollNumber: { type: new GraphQLNonNull(GraphQLString) },
+    contact: { type: new GraphQLNonNull(GraphQLString) },
+    email: { type: new GraphQLNonNull(GraphQLString) },
+    prefect: { type: new GraphQLNonNull(GraphQLString) },
+  },
+  async resolve(parent, { name, rollNumber, contact, email, prefect }) {
+    try {
+      const checkMentor = await Mentor.exists({ rollNumber });
```
Comment (abhibhaw) [inline server/schema/mutations/mentor.js:25]: Same as above

File: server/schema/mutations/mentor.js [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+};
+
+const removeMentor = {
+  type: MentorType,
+  args: {
+    rollNumber: { type: new GraphQLNonNull(GraphQLString) },
+  },
+  async resolve(parent, { rollNumber }) {
+    await Mentor.findOneAndDelete({ rollNumber }, (error, docs) => {
+      if (error) {
```
Comment (abhibhaw) [inline server/schema/mutations/mentor.js:93]: same as above

Comment (abhibhaw) [inline server/schema/mutations/mentor.js:93]: same as above

File: server/schema/mutations/prefect.js [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+  args: {
+    name: { type: new GraphQLNonNull(GraphQLString) },
+    rollNumber: { type: new GraphQLNonNull(GraphQLString) },
+    contact: { type: new GraphQLNonNull(GraphQLString) },
+    email: { type: new GraphQLNonNull(GraphQLString) },
+    year: { type: GraphQLInt },
+  },
+  async resolve(parent, { name, contact, email, coordinator, rollNumber }) {
+    try {
+      const checkPrefect = await Prefect.exists({ rollNumber });
```
Comment (abhibhaw) [inline server/schema/mutations/prefect.js:25]: Same as above

File: server/schema/mutations/prefect.js [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+  },
+};
+
+const removePrefect = {
+  type: PrefectType,
+  args: {
+    rollNumber: { type: new GraphQLNonNull(GraphQLString) },
+  },
+  async resolve(parent, { rollNumber }) {
+    await Prefect.findOneAndDelete({ rollNumber }, (error, docs) => {
```
Comment (abhibhaw) [inline server/schema/mutations/prefect.js:89]: Same as above

File: server/schema/queries.js [hunk 10]
```diff
@@ -1,51 +1,82 @@
...
+      type: new GraphQLList(MenteeType),
+      args: {},
+      async resolve() {
+        return Prefect.find({});
+      },
+    },
+    getCoordinators: {
+      type: new GraphQLList(MenteeType),
+      args: {},
+      async resolve() {
```
Comment (abhibhaw) [inline server/schema/queries.js:75]: should we fetch only `isActive` ones

**Review by abhibhaw — CHANGES_REQUESTED** [review]

**Review by abhibhaw — APPROVED** [review]
lgtm

File: server/schema/mutations/coordinator.js [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+          },
+        },
+        {
+          omitUndefined: true,
+          new: true,
+          runValidators: true,
+        }
+      );
+
+      return await coordinator.save();
```
Comment (DesignrKnight) [inline server/schema/mutations/coordinator.js:70]: Are you sure about this? findOneAndUpdate and then calling save again, This seems wrong

File: server/schema/queries.js [hunk 12]
```diff
@@ -1,51 +1,85 @@
...
       },
     },
-    coordinators: {
+    getCoordinatorByRollnumber: {
       type: new GraphQLList(CoordinatorType),
-      args: {},
+      args: { rollNumber: { type: GraphQLString } },
       resolve(parent, args) {
-        return Coordinator.find({});
+        return Coordinator.find({
```
Comment (DesignrKnight) [inline server/schema/queries.js:46]: is a filter for `isActive `required when doing a `findOne `on a unique field? Having `isActive `filter on expecting array of response is okay, but why that filter when `findOne`?

**Pull request merged**